First principles datasets

gAldeia opened the pull request:
Data comes from two symbolic regression repos:
- Miles Cranmer's PySR: https://github.com/MilesCranmer/PySR
- Etienne Russeil et al.'s MvSR: https://github.com/erusseil/MvSR-analysis

They are all datasets that have a first-principle equation derived from data and used in their respective papers to show how symbolic regression has the potential of retrieving the original equation when only observational data is available.

While some of them have just a few samples and  others are synthetically generated, they are challenging for symbolic regression methods and can be used to evaluate these algorithms.

The idea of pushing them into PMLB is to help other users to quickly set up experiments with the data.

I still need to write proper metadata for them. My understanding is that opening a PR will trigger a GA that will push some new files to my fork, which I should complete before the new datasets go to revision. Please let me know if there is there anything I got wrong and need to update!

gAldeia committed:
First principles datasets

Data comes from two symbolic regression repos:
- Miles Cranmer's PySR: https://github.com/MilesCranmer/PySR
- Etienne Russeil et al.'s MvSR: https://github.com/erusseil/MvSR-analysis

They are all datasets that have a first-principle equation
derived from data and used in their respective papers
to show how symbolic regression has the potential of retrieving
the original equation when only observational data is available.

While some of them have just a few samples and  others are synthetically
generated, they are challenging for symbolic regression methods and
can be used to evaluate these algorithms.

The idea of pushing them into PMLB is to help other users to
quickly set up experiments with the data.

I still need to write proper metadata for them.

gAldeia committed:
Re-generated broken datasets

CI was failing to parse the contents of these specific ones.

github-actions[bot] committed:
update dataset files

Created by https://github.com/gAldeia/pmlb/actions/runs/11616806556\nfrom f23672c on 2024-10-31

trangdata commented:
Thank you for this PR, @gAldeia! 💯  🌈 

If you could update the metadata.yaml files (e.g. [datasets/first_principles_rydberg/metadata.yaml](https://github.com/EpistasisLab/pmlb/pull/181/files#diff-1b1d8c327021e7985f07ff274d75f1c4ba04fc1e5a62ee9b589e2476616f6e0a)), I will review!

gAldeia committed:
New metadata

gAldeia committed:
Updated summary

gAldeia committed:
Merge branch 'symreg_first_principles' of https://github.com/gAldeia/pmlb into symreg_first_principles

gAldeia commented:
Done! Let me know if everything is ok or if it needs a major review.
Thanks!

github-actions[bot] committed:
update dataset files

Created by https://github.com/gAldeia/pmlb/actions/runs/13434894123\nfrom bdc87c8 on 2025-02-20

gAldeia committed:
Merge branch 'EpistasisLab:master' into symreg_first_principles

gAldeia reviewed (commented):
This is correct, I accidentally pushed changes from another branch into this one

trangdata commented:
ah I think one of the datasets' names are not matching (directory name vs. metadata dataset field): 
- first_principles_supernovae_z**r**
- first_principles_supernovae_z**g**
Could you fix plz @gAldeia

gAldeia committed:
Fix typo

gAldeia committed:
Merge branch 'symreg_first_principles' of https://github.com/gAldeia/pmlb into symreg_first_principles

trangdata committed:
Merge branch 'master' into symreg_first_principles

github-actions[bot] committed:
update dataset files

Created by https://github.com/gAldeia/pmlb/actions/runs/13465733857\nfrom a226e6b on 2025-02-21
